compiler/src/iree/compiler/Dialect/Stream/IR/StreamOps.td

```diff
@@ -216,11 +216,17 @@ def Stream_ResourceDeallocaOp : Stream_Op<"resource.dealloca", [
     The resource is considered live and valid until the provided timepoint is
     reached and the memory is only made available for future requests after
     the result timepoint is reached.
+
+    If `prefer_origin` is set then the deallocation will be scheduled against
+    the original placement affinity of the allocation at runtime. In cases where
+    the original placement is unavailable the affinity of the deallocation op
+    will be used to perform a barrier operation.
```

Comment:
Why is the barrier needed here?

Reply:
We always need a way to preserve the wait -> signal and may not know until runtime that an operation is a no-op - if it is a no-op we've already allocated unique fences and don't have a way to alias them. This way we can preserve the timeline as-if a deallocation takes place even if one doesn't (for whatever reason).